compiler/rustc_session/src/filesearch.rs

```diff
@@ -132,6 +132,23 @@ fn current_dll_path() -> Result<PathBuf, String> {
         .clone()
 }
 
+#[cfg(target_os = "cygwin")]
```

Comment:
I'm not really up on the rust specifics here, but is there a reason this is an entirely separate function for `cygwin` rather than being more merged into the above `unix` case?  The only difference should be that `dli_fname` is an array of char on cygwin while generally it's a pointer to char.  In C those are accessed the same way, but apparently in rust you need an extra `.as_ptr()`, could that be done in a more minimal way?

Reply:
You're right. It's simpler now.